planner/optimizer/src/main/java/eu/seaclouds/platform/planner/optimizer/CloudOffer.java

```diff
@@ -91,7 +97,8 @@ public void setCost(double cost) {
 
    @Override
    public CloudOffer clone() {
```

Comment:
In my opinion, in order to clone objects in Java maybe is better to use the native interface to perform this (http://stackoverflow.com/questions/869033/how-do-i-copy-an-object-in-java) also watch out with the member variables if any of them are references to objects (like lists) and you don't clone the list but the reference (it works with strings because in Java strings are immutable, but just in case take this into consideration)